java/pattern/variable/BoundVariable.java

```diff
@@ -42,7 +41,7 @@ public abstract class BoundVariable extends Variable implements Conjunctable {
 
     @Override
     public void validateIsBoundedBy(Set<UnboundVariable> bounds) {
-        if (Stream.concat(Stream.of(this), variables()).noneMatch(v -> bounds.contains(v.toUnbound()))) {
+        if (Stream.concat(Stream.of(this), variables()).map(BoundVariable::toUnbound).noneMatch(v -> bounds.contains(v))) {
```

Comment:
this line (and other validations elsewhere) no longer need to change right? you can just use the old code?

Reply:
Yes